lib/elixir/lib/map.ex

```diff
@@ -565,9 +568,11 @@ defmodule Map do
   """
   @spec pop(map, key, value) :: {value, map}
   def pop(map, key, default \\ nil) do
-    case map do
-      %{^key => value} -> {value, delete(map, key)}
-      %{} -> {default, map}
+    case fetch(map, key) do
+      {:ok, value} ->
+        {value, delete(map, key)}
+      :error ->
+        {default, map}
```

Comment:
@michalmuskala we would do that in a few places so we decided to go with `fetch/2` in order to ensure that's always done.